### PR TITLE
Bring the storage blob SDK version to match the latest released version.

### PR DESCRIPTION
**Pull request by pallavit**
# Description

Bring the storage blob SDK version to match the latest released version.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/versioning/version_client.txt [hunk 1]
```diff
@@ -121,7 +121,7 @@ com.azure:azure-security-keyvault-perf;1.0.0-beta.1;1.0.0-beta.1
 com.azure:azure-sdk-template;1.2.1-beta.2;1.2.1-beta.16
 com.azure:azure-spring-data-cosmos;3.14.0;3.15.0-beta.1
 com.azure:azure-spring-data-cosmos-test;3.0.0-beta.1;3.0.0-beta.1
-com.azure:azure-storage-blob;12.14.1;12.15.0-beta.2
+com.azure:azure-storage-blob;12.14.2;12.15.0-beta.2
```
Comment (srnagar) [inline eng/versioning/version_client.txt:124]: This version update usually is done through the autogenerated PR. Was the automated PR to increment the version not created?

**Review by srnagar — APPROVED** [review]

File: eng/versioning/version_client.txt (hunk 1 above)
Comment (g2vinay) [reply eng/versioning/version_client.txt:124]: release was done via release branch.

**Review by g2vinay — APPROVED** [review]

File: eng/versioning/version_client.txt (hunk 1 above)
Comment (pallavit) [reply eng/versioning/version_client.txt:124]: These libraries were released from the release branch. The manual PR that was backported to main missed this library.

**Review by rickle-msft — APPROVED** [review]

**Pull request merged**